File: DesktopCore/src/burlton/desktopcore/code/screen/LoadingDialog.java
package burlton.desktopcore.code.screen;

import java.awt.BorderLayout;

import javax.swing.JDialog;
import javax.swing.JLabel;
import javax.swing.SwingConstants;
import javax.swing.SwingUtilities;
import javax.swing.WindowConstants;

public class LoadingDialog extends JDialog
{
	public LoadingDialog()
	{
		setSize(200, 100);
		setLocationRelativeTo(null);
		setResizable(false);
		setModal(true);
		setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
		lblMessage.setHorizontalAlignment(SwingConstants.CENTER);
		getContentPane().add(lblMessage, BorderLayout.CENTER);
	}
	
	private final JLabel lblMessage = new JLabel("Communicating with Server...");
	
	public void showDialog(final String textToShow)
	{
		Runnable showRunnable = new Runnable()
		{
			@Override
			public void run()
			{
				lblMessage.setText(textToShow);
				setVisible(true);
			}
		};
		
		SwingUtilities.invokeLater(showRunnable);
	}
	
	public void dismissDialog()
	{
		Runnable hideRunnable = new Runnable()
		{
			@Override
			public void run()
			{
				setVisible(false);
			}
		};
		
		SwingUtilities.invokeLater(hideRunnable);
	}
}

File: Darts/src/burlton/dartzee/code/core/screen/LoadingDialog.kt
package burlton.dartzee.code.core.screen

import net.miginfocom.swing.MigLayout
import javax.swing.*

class LoadingDialog : JDialog()
{
    private val lblMessage = JLabel("Communicating with Server...")

    init
    {
        layout = MigLayout("gapy 12px", "[grow]", "[]")
        setSize(250, 115)
        setLocationRelativeTo(null)
        isResizable = false
        isModal = true
        isAlwaysOnTop = true
        defaultCloseOperation = WindowConstants.DO_NOTHING_ON_CLOSE
        lblMessage.horizontalAlignment = SwingConstants.CENTER
        contentPane.add(lblMessage, "cell 0 0, alignx center")

        val progressBar = JProgressBar()
        progressBar.isIndeterminate = true

        contentPane.add(progressBar, "cell 0 1, alignx center")
    }

    fun showDialog(textToShow: String)
    {
        val showRunnable = Runnable {
            lblMessage.text = textToShow
            isVisible = true
        }

        SwingUtilities.invokeLater(showRunnable)
    }

    fun dismissDialog()
    {
        val hideRunnable = Runnable { isVisible = false }

        SwingUtilities.invokeLater(hideRunnable)
    }
}